test/retry_policy_tests.py

```diff
@@ -275,6 +275,38 @@ def test_default_retry_policy_for_create(self):
 
         retry_utility._ExecuteFunction = self.OriginalExecuteFunction
 
+    def test_429_on_first_page(self):
+        client = document_client.DocumentClient(Test_retry_policy_tests.host, {'masterKey': Test_retry_policy_tests.masterKey})
+
+        document_definition = { 'id': 'doc429',
+                                'name': 'sample document',
+                                'key': 'value'}
+
+        created_document = client.CreateDocument(self.created_collection['_self'], document_definition)
+
+        # Mock an overloaded server which always returns 429 Too Many
+        # Requests, by hooking the client's POST method.
+        original_post_function = client._DocumentClient__Post
+        client._DocumentClient__Post = self._MockPost429
+
+        # Test: query for the document.  Expect the mock overloaded server
+        # to raise a 429 exception.
+        try:
+            query = client.QueryDocuments(self.created_collection['_self'], "SELECT * FROM c")
+            docs = list(query) # force execution now
+            self.assertFalse(True, 'function should raise HTTPFailure.')
+
+        except errors.HTTPFailure as ex:
+            self.assertEqual(ex.status_code, StatusCodes.TOO_MANY_REQUESTS)
+
+        client._DocumentClient__Post = original_post_function
```

Comment:
Mocking out the server returning 429 error by overriding the clients `__Post` method.  It needs to be done this low down, because mocking at the `_Execute` level (as in the rest of this file) misses out the lower level of retries.